addons/website_quote/controllers/main.py

```diff
@@ -18,50 +15,53 @@ class sale_quote(http.Controller):
-            report_obj = request.registry['report']
-            pdf = report_obj.get_pdf(request.cr, SUPERUSER_ID, [order_id], 'website_quote.report_quote', data=None, context=request.context)
+            pdf = request.env['report'].sudo().get_pdf(order, 'website_quote.report_quote')
             pdfhttpheaders = [('Content-Type', 'application/pdf'), ('Content-Length', len(pdf))]
             return request.make_response(pdf, headers=pdfhttpheaders)
-        user = request.registry['res.users'].browse(request.cr, SUPERUSER_ID, request.uid, context=request.context)
-        tx_id = request.registry['payment.transaction'].search(request.cr, SUPERUSER_ID, [('reference', '=', order.name)], context=request.context)
-        tx = request.registry['payment.transaction'].browse(request.cr, SUPERUSER_ID, tx_id, context=request.context) if tx_id else False
+        user_sudo = request.env.user.sudo()
+        payment_transaction = request.env['payment.transaction'].sudo().search([('reference', '=', order.name)])
         values = {
             'quotation': order,
             'message': message and int(message) or False,
-            'option': bool(filter(lambda x: not x.line_id, order.options)),
-            'order_valid': (not order.validity_date) or (now <= order.validity_date),
-            'days_valid': days,
-            'action': action,
-            'breadcrumb': user.partner_id == order.partner_id,
-            'tx_id': tx_id,
-            'tx_state': tx.state if tx else False,
-            'tx_post_msg': tx.acquirer_id.post_msg if tx else False,
-            'need_payment': not tx_id and order.state == 'manual'
+            'option': bool(order.options.filtered(lambda x: not x.line_id)),
+            'order_valid': (not order.validity_date) or (now <= fields.Date.from_string(order.validity_date)),
+            'days_valid': max(days, 0),
+            'action': action_id,
+            'breadcrumb': user_sudo.partner_id == order.partner_id,
+            'tx_id': payment_transaction.id,
+            'tx_state': payment_transaction.state,
+            'tx_post_msg': payment_transaction.acquirer_id.post_msg,
+            'need_payment': not payment_transaction and order.state == 'manual'
         }
 
-        if order.require_payment or (not tx_id and order.state == 'manual'):
-            payment_obj = request.registry.get('payment.acquirer')
-            acquirer_ids = payment_obj.search(request.cr, SUPERUSER_ID, [('website_published', '=', True), ('company_id', '=', order.company_id.id)], context=request.context)
-            values['acquirers'] = list(payment_obj.browse(request.cr, token and SUPERUSER_ID or request.uid, acquirer_ids, context=request.context))
-            render_ctx = dict(request.context, submit_class='btn btn-primary', submit_txt=_('Pay & Confirm'))
+        if order.require_payment or (not payment_transaction and order.state == 'manual'):
+            PaymentAcquirer = request.env['payment.acquirer']
+            if token:
```

Comment:
check read access on payment.acquirer, by default all user can read it, so i think no need to check an use sudo